docs/examples/junit4/generic/src/test/java/generic/ImageNameSubstitutionTest.java

```diff
@@ -0,0 +1,43 @@
+package generic;
+
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class ImageNameSubstitutionTest {
+
+    @Test
+    public void simpleExample() {
+        try (
+            // directDockerHubReference {
+            // Referring directly to an image on Docker Hub (mysql:8.0.22)
+            final MySQLContainer<?> mysql = new MySQLContainer<>(
+                DockerImageName.parse("mysql:8.0.22")
+            )
+
+            // start the container and use it for testing
+            // }
+        ) {
+            mysql.start();
+        }
+    }
+
+    @Test @Ignore
```

Comment:
just a random thought:

can't we make this test pass by substituting `registry.mycompany.com/mirror/mysql` with `mysql` (aka reverse substitution)?

Reply:
Very good idea - will do.

Reply:
Done